content/content-tool/tool/src/java/org/sakaiproject/content/entityproviders/ContentEntityProvider.java

```diff
@@ -179,6 +181,10 @@ public boolean entityExists(String id) {
 						ContentItem item = new ContentItem();
 						item.setType("collection");
 						item.setSize(contentHostingService.getCollectionSize(id));
+						if (allowUpdateSite) // to be consistent with UI
+							item.setQuota(Long.toString(contentHostingService.getQuota(collection)));
+						item.setUsage(Long.toString(collection.getBodySizeK() * 1024));
```

Comment:
The quota usage looks like it should always be fast as the columns that pre-calculate it has to exist.

`org.sakaiproject.content.impl.BaseContentService.BaseCollectionEdit#getBodySizeK()`